### PR TITLE
intel: consolidate libs() in the base class

### PR DESCRIPTION
**Pull request by davydden**
follow up to https://github.com/spack/spack/pull/10993 , in that PR I did not notice that the base class already has `libs()`, we should have put everything there from the beginning.

### CODE REVIEW

File: lib/spack/spack/build_systems/intel.py [hunk 1]
```diff
@@ -935,18 +935,25 @@ def headers(self):
...
+        if '+mkl' in self.spec or self.provides('lapack'):
+            result = self.lapack_libs + result
         if '+mpi' in self.spec or self.provides('mpi'):
             # If prefix is too general, recursive searches may get files from
             # supported but inappropriate sub-architectures like 'mic'.
             libnames = ['libmpifort', 'libmpi']
             if 'cxx' in self.spec.last_query.extra_parameters:
                 libnames = ['libmpicxx'] + libnames
-            result += find_libraries(
+            result = find_libraries(
```
Comment (healther) [inline lib/spack/spack/build_systems/intel.py]: why the change of `+=` to `= ... +`?

Comment (davydden) [reply lib/spack/spack/build_systems/intel.py]: For consistency with the rest of the function (always prepend). Although all providers up to this point did not depend on `mpi`, so it won't matter for this line.
